src/mulsi/processor.py

```diff
@@ -0,0 +1,58 @@
+"""Preprocessing functions for MULSI.
+"""
+
+import einops
+import torch
+from tensordict import TensorDict
+
+
+class DiffClipProcessor:
+    """Differentiable CLIP processor.
+
+    TODO: Temporary solution. Should be inheriting from the processor.
+    """
```

Comment:
This class is specific for CLIP? You're planning on having an abstract class for processors (just like for attacks) ?

Reply:
Unsure about the abstraction here (maybe with tokenizer), but it might be integrated with `MulsiModel`.

Like `MulsiModel` = `PreTainedModel` + `Processor` + `Losses` (can have multiple targets)